ArduCopter/mode_auto.cpp

```diff
@@ -408,6 +415,81 @@ void ModeAuto::circle_start()
     }
 }
 
+Vector3f Mode::_LoiterTurns::get_circle_edge_neu() const
+{
+    Vector3f circle_edge_neu;
+    copter.circle_nav->get_closest_point_on_circle(circle_edge_neu);
+    return circle_edge_neu;
+}
+
+// check if the vehicle needs moving to the edge of cirlce
+bool Mode::_LoiterTurns::needs_moving_to_edge()
+{
+    if (get_state() != _LoiterTurns::State::READY) {
+        // should not have been called
+        return false;
+    }
+
+    // check our distance from edge of circle
+    const float dist_to_edge = (copter.inertial_nav.get_position_neu_cm() - get_circle_edge_neu()).length();
+    return dist_to_edge > 300.0f;
```

Comment:
```suggestion
    const float dist_to_edge = (copter.inertial_nav.get_position_neu_cm() - get_circle_edge_neu()).length_squared();
    return dist_to_edge > 300.0f*300.0f;
```

taking the square root of a number is really rather expensive.

Reply:
Done